haystack/nodes/other/route_documents.py

```diff
@@ -11,8 +11,8 @@ class RouteDocuments(BaseComponent):
     different nodes.
     """
 
-    # By default (split_by == "content_type"), the node has two outgoing edges.
-    outgoing_edges = 2
+    # As this is dynamic, set it to a high number for static pipeline validation
+    outgoing_edges = 999
```

Comment:
Maybe worth documenting? I'm not sure either, but because I'm in doubt, I'd say we should :sweat_smile: